pkg/cluster/config/v1alpha3/types.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/kind/pkg/container/cri"
+	"sigs.k8s.io/kind/pkg/kustomize"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Cluster contains kind cluster configuration
+type Cluster struct {
+	// TypeMeta representing the type of the object and its API schema version.
+	metav1.TypeMeta `json:",inline"`
+
+	// Nodes contains the list of nodes defined in the `kind` Cluster
+	Nodes []Node `json:"nodes"`
+
+	/* Advanced fields */
+
+	// KubeadmConfigPatches are applied to the generated kubeadm config as
+	// strategic merge patches to `kustomize build` internally
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/strategic-merge-patch.md
+	// This should be an inline yaml blob-string
+	KubeadmConfigPatches []string `json:"kubeadmConfigPatches,omitempty"`
+
+	// KubeadmConfigPatchesJSON6902 are applied to the generated kubeadm config
+	// as patchesJson6902 to `kustomize build`
+	KubeadmConfigPatchesJSON6902 []kustomize.PatchJSON6902 `json:"kubeadmConfigPatchesJson6902,omitempty"`
+}
+
+// Node contains settings for a node in the `kind` Cluster.
+// A node in kind config represent a container that will be provisioned with all the components
+// required for the assigned role in the Kubernetes cluster
+type Node struct {
+	// Role defines the role of the node in the in the Kubernetes cluster
+	// created by kind
+	//
+	// Defaults to "control-plane"
+	Role NodeRole `json:"role,omitempty"`
+
+	// Image is the node image to use when creating this node
+	Image string `json:"image,omitempty"`
+
+	/* Advanced fields */
+
+	// ExtraMounts describes additional mount points for the node container
+	// These may be used to bind a hostPath
+	ExtraMounts []cri.Mount `json:"extraMounts,omitempty"`
+}
+
+// NodeRole defines possible role for nodes in a Kubernetes cluster managed by `kind`
+type NodeRole string
+
+const (
+	// ControlPlaneRole identifies a node that hosts a Kubernetes control-plane.
+	//
+	// NOTE: in single node clusters, control-plane nodes act also as a worker nodes
```

Comment:
possibly use two lines only:
```
// ControlPlaneRole identifies a node that hosts a Kubernetes control-plane.
// NOTE: in single node clusters, control-plane nodes will allow workloads.
```
^ proposed minor change to the note.

Reply:
sure, SGTM

Reply:
done